compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -450,6 +454,23 @@ impl<'a, 'b, 'tcx> AssocTypeNormalizer<'a, 'b, 'tcx> {
             value.fold_with(self)
         }
     }
+
+    fn fold_reveal(&mut self, ty: Ty<'tcx>, def_id: DefId, substs: SubstsRef<'tcx>) -> Ty<'tcx> {
```

Comment:
This should have a more descriptive name like `fold_alias_or_revealed_opaque`, and maybe a comment about how it's a helper for those two cases.